welfare/views.py

```diff
@@ -160,6 +171,44 @@ def _get_percent(with_services: list, total: list) -> float:
     return with_services / total * 100
 
 
+def _get_service_detail_list(students) -> list:
+    if not students:
+        return []
+    results = list()
+
+    all_students_count = len(students)
+    all_girls_count = len([ s for s in students if s.sex.upper() == "FEMALE" ])
+    all_boys_count = len([ s for s in students if s.sex.upper() == "MALE" ])
+
+    for service in SupportService.objects.all():
```

Comment:
consider
```
total_count = service.studentsupportassoc_set.count()
boy_count = service.studentsupportassoc_set.filter(student__sex='male')
girl_count = service.studentsupportassoc_set.filter(student__sex='female')
```

Instead of the for looping over students (__ double underscore does a join between tables)

Reply:
A given student can be associated multiple times with the same service, and vice versa, e.g. over different date "windows" (with differing comments).  Because of this, I think the suggested approach would result in some students being counted multiple times.